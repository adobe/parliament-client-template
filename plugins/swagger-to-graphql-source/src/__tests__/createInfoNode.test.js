const createInfoNode = require("../createInfoNode")

const SwaggerParser = require("@apidevtools/swagger-parser")

const path = require("path")

const DATA_PATH = path.resolve(__dirname, "data/swagger.json")

const parentFile = {
  relativePath: "parentFile.relativePath",
  id: "parentFile.id",
}
const createNodeId = jest.fn(data => data)
const createContentDigest = jest.fn(data => "HashString")

const gatsbyApi = {
  createNodeId,
  createContentDigest,
}

test("create info node", async () => {
  const swaggerObject = await SwaggerParser.parse(DATA_PATH)

  const props = { swaggerObject, parentFile, gatsbyApi }

  const nodeData = createInfoNode(props)

  expect(nodeData).toMatchSnapshot()
})
