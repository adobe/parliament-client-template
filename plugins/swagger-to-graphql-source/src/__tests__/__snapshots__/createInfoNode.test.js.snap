// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create info node 1`] = `
Object {
  "basePath": "/v2",
  "children": Array [],
  "consumes": undefined,
  "contact": Object {
    "email": "apiteam@swagger.io",
  },
  "description": "This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key \`special-key\` to test the authorization filters.",
  "host": "petstore.swagger.io",
  "id": "parentFile.relativePath Swagger Petstore 1.0.5",
  "internal": Object {
    "content": "",
    "contentDigest": "HashString",
    "type": "SwaggerOpenApiInfo",
  },
  "license": Object {
    "name": "Apache 2.0",
    "url": "http://www.apache.org/licenses/LICENSE-2.0.html",
  },
  "name": "parentFile.relativePath",
  "parent": "parentFile.id",
  "produces": undefined,
  "schemes": Array [
    "https",
    "http",
  ],
  "swagger": "2.0",
  "tags": Array [
    Object {
      "description": "Everything about your Pets",
      "externalDocs": Object {
        "description": "Find out more",
        "url": "http://swagger.io",
      },
      "name": "pet",
    },
    Object {
      "description": "Access to Petstore orders",
      "name": "store",
    },
    Object {
      "description": "Operations about user",
      "externalDocs": Object {
        "description": "Find out more about our store",
        "url": "http://swagger.io",
      },
      "name": "user",
    },
  ],
  "termsOfService": "http://swagger.io/terms/",
  "title": "Swagger Petstore",
  "version": "1.0.5",
}
`;
